### PR TITLE
Make the FloatFuncs trait a sealed public trait, add sin/sinf.

### PR DESCRIPTION
**Pull request by ratmice**
This trait, with the addition of sin/sinf is needed to build peniko with the `--no-default-features --features libm` combination (added in the pr below).

The approach taken here is that rather than duplicating this common module in peniko, to make it pub, and add the sin/sinf. for peniko.  This trait is only conditionally defined so use of it in peniko would look something like:

```
+#[cfg(not(feature = "std"))]
+#[allow(unused_imports)]
+use kurbo::common::FloatFuncs as _
```

This patch is in regards to fixing the testsuite failures for default features added in
https://github.com/linebender/peniko/pull/23

### CODE REVIEW

File: src/common.rs
Comment (DJMcNab) [inline src/common.rs:8]: I'm not sure that `common_sealed` is a great name for this - I think I'd prefer something like:
```rust
mod sealed {
    /// Seals [super::FloatFunc]
    pub trait FloatFuncsSealed {}
}
```

I don't think the Rust project documents sealed traits itself, so I think we should also probably have a link which explains it.
I don't know what link that would be, however

File: src/common.rs
Comment (DJMcNab) [inline src/common.rs:21]: This should have some docs now that it's public

**Review by DJMcNab — COMMENTED** [review]
I'm not a Kurbo maintainer, so I'm not certain that they would want this

File: src/common.rs
Comment (ratmice) [reply src/common.rs:6]: It looks like this is here to avoid documenting the existing `pub const GAUSS_LEGENDRE_COEFFS`* values.
I added basically a stub documentation explaining the trait, and pointing to the std implementation for further docs.

File: src/common.rs
Comment (ratmice) [reply src/common.rs]: fixed in [cb3c29d](https://github.com/linebender/kurbo/pull/332/commits/cb3c29d41a0163f3118954e4c0ef44a0b53023f2)

File: src/common.rs
Comment (ratmice) [reply src/common.rs]: Should be fixed in [85ad5eb](https://github.com/linebender/kurbo/pull/332/commits/85ad5eb179ebcf2c6c28f775ab482a8143f223f8)

**Comment by ratmice** [comment]
One thing this patch *doesn't* try to fix is `cargo test --no-default-features --features libm` I manually checked that the respective `build` target works, don't know see that this combination is actually being tested by ci though.

It is perhaps worth adding that.

File: src/common.rs
Comment (DJMcNab) [inline src/common.rs:9]: I think this reads:
A sealed trait which sealed trait which..., which doesn't seem right

Comment (ratmice) [reply src/common.rs]: whoops.

**Comment by derekdreery** [comment]
Hey @ratmice sorry if this is a silly question, but why do we need to seal the trait?

**Comment by ratmice** [comment]
I just did it in the spirit of it being the more conservative change, i.e. it is easy to unseal it later, but sealing it would be a breaking change.

So honestly no *good* reason.

**Comment by derekdreery** [comment]
> I just did it in the spirit of it being the more conservative change, i.e. it is easy to unseal it later, but sealing it would be a breaking change.

@ratmice Would you mind putting a comment on the sealing trait along these lines? Then future me knows we can remove the sealing if we need to :). Other than that LGTM.

File: src/common.rs
Comment (DJMcNab) [inline src/common.rs:11]:
```suggestion
    /// A [sealed trait](https://predr.ag/blog/definitive-guide-to-sealed-traits-in-rust/)
    /// which stops [super::FloatFuncs] from being implemented outside kurbo. This could be relaxed
    /// in the future if there are good reasons to allow outside impls.
```
I'd also probably change 'outside' to 'external' in the final line

Comment (ratmice) [reply src/common.rs]: Should be fixed in [4684dac](https://github.com/linebender/kurbo/pull/332/commits/4684dacad8fd2fb5929a510364c9196ae5c7ec32)

Comment (ratmice) [reply src/common.rs]: I think this link fails with clippy because of the `#[cfg(not(features="std")))]` conditional compilation?

**Comment by ratmice** [comment]
If it's alright i'll squash/force push this all down to one commit with the initial commit message.

**Comment by DJMcNab** [comment]
That would be alright, although that will also happen as part of the merge process

**Review by derekdreery — APPROVED** [review]
LGTM

**Comment by ratmice** [comment]
I'm happy with it as is, and ran the peniko patch on top of this revision through it's CI.

I don't believe anything in a peniko review would require changes here, but feel like I
should wait for at least a cursory a review there until landing this here.

**Pull request merged**
